src/test_l3_dsFPD.c

```diff
@@ -0,0 +1,692 @@
+    UT_LOG_INFO("Result dsSetFPColor() dsError_t:[%s]",UT_Control_GetMapString(dsFrontPanelErrorCodeTable, status));
+    assert(status == dsERR_NONE);
+    exit:
+    UT_LOG_INFO("Out %s\n", __FUNCTION__);
+}
+/**
+* @brief This test provides a scope to get Color of the Indicator set.
+*
+* This test case provides a scope to  get the Color of Given indicator
+*
+* **Pre-Conditions:** @n
+* Front panel Module should be intialized through Test 1 before calling this test.
+* Front panel indicator state should be set to ON through Test 2 before calling this test
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select the Test 10 and provide indicator index.
+*
+*/
+void test_l3_dsFPD_hal_GetFPColor(void)
+{
+    gTestID = 11;
+    UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t status = dsERR_NONE;
+    dsFPDColor_t color;
+    int32_t ret = 0;
+
+    ret = selectIndicator();
+    if(!ret)
+        goto exit;
+    UT_LOG_INFO("Calling dsGetFPColor(IN:Indicator:[0x%d], OUT:Color)",eIndicator);
+    status = dsGetFPColor((dsFPDIndicator_t)eIndicator,&color);
+    UT_LOG_INFO("Result dsGetFPColor(IN:Indicator:[0x%d], OUT:Color:[%d]) dsError_t:[%s]" \
+                                                ,eIndicator,color,UT_Control_GetMapString(dsFrontPanelErrorCodeTable, status));
+    assert(status == dsERR_NONE);
+    /* Check that the Indicator is valid */
+    exit:
+    UT_LOG_INFO("Out %s\n", __FUNCTION__);
+}
+/**
+* @brief This test provides a scope to Termnate Device Settings Front panel Module.
+*
+* This test case provides a scope to terminate front panel module.
+*
+* **Pre-Conditions:** @n
+* Front panel Module should be intialized through Test 1 before calling this test.
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select the Test 11.
+*
+*/
+
+void test_l3_dsFPD_hal_Term(void)
+{
+    gTestID = 12;
+    dsError_t status = dsERR_NONE;
+
+    UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    // Step 1: Call dsFPInit()
+    UT_LOG_INFO("Calling dsFPTerm()");
+    status = dsFPTerm();
+    UT_LOG_INFO("Result dsFPTerm()dsError_t:[%s]",UT_Control_GetMapString(dsFrontPanelErrorCodeTable, status));
+    assert(status == dsERR_NONE);
+
+    UT_LOG_INFO("Out %s\n", __FUNCTION__);
+}
+
+static UT_test_suite_t * pSuite = NULL;
+
+/**
+ * @brief Register the main tests for this module
+ *
+ * @return int32_t - 0 on success, otherwise failure
+ */
+int32_t test_l3_dsFPD_register(void)
+{
+    // Create the test suite
+    pSuite = UT_add_suite("[L3 Front Panel Functions] ", NULL, NULL);
+    if (pSuite == NULL)
+    {
+        return -1;
+    }
+    // List of test function names and strings
+
+    UT_add_test( pSuite, "1_Init_dsFPD", test_l3_dsFPD_hal_Init);
+    UT_add_test( pSuite, "2_SetState_dsFPD", test_l3_dsFPD_hal_SetFPState);
```

Comment:
Why added number in the functionality name? Is any requested? All other modules are following the functionality name without the numbers

Reply:
updated to latest format.